### PR TITLE
Instrument mod_event_pusher

### PR DESCRIPTION
**Pull request by gustawlippa**
This PR addresses instruments mod_event_pusher. I didn't touch `mod_event_pusher_rabbit`, because it reports `mongoose_rabbit_worker`, and it's better to do them together with other workers' metrics.

`mod_event_pusher_http` treats any HTTP reason code as a success, and I didn't change this behaviour.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/219719) / elasticsearch_and_cassandra_mnesia / c83a9b0715054d5baff36669a18d063d9193f704
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4285/219719/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4285/219719/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40641f31a4ef17.2024-05-22_08.56.08/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

File: src/event_pusher/mod_event_pusher_http.erl [hunk 1]
```diff
@@ -113,33 +114,19 @@ make_req(Acc, Dir, Domain, Sender, Receiver, Message, Opts) ->
...
-    ok;
-record_result(HostType, {error, Reason}, _, LogMeta) ->
-    mongoose_metrics:update(HostType, ?FAILED_METRIC, 1),
+    mongoose_instrument:span(?SENT_METRIC, #{host_type => HostType},
+                             fun mongoose_http_client:post/5, [HostType, PoolName, Path, Headers, Body],
+                             fun(Time, Result) -> handle_post_result(Time, Result, LogMeta) end).
+
+handle_post_result(Time, {ok, {CodeBin, _Body}}, #{sender := Sender}) ->
+    % Fusco returns HTTP response codes as binaries
+    Code = binary_to_integer(CodeBin),
```
Comment (chrzaszcz) [inline src/event_pusher/mod_event_pusher_http.erl:123]: The only problem I see is that fusco seems to just return the three characters, while some misbehaving server might return a non-numerical status code. In this case, maybe instrumentation should't fail? Maybe we could return the binary?

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general. I added a comment. Regarding the `rabbit` worker - I agree that we could do it separately. It's a bit weird, because it looks like for other Rabbit pools the metrics are not set up... we could unify this when reworking pools.

File: src/event_pusher/mod_event_pusher_http.erl (hunk 1 above)
Comment (gustawlippa) [reply src/event_pusher/mod_event_pusher_http.erl:123]: I also wasn't sure how to tackle it. I will change this to return the binary, as indeed the instrumentation shouldn't fail.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/220073) / elasticsearch_and_cassandra_mnesia / bbc8ff2d8fc98f72bdd77106933b57a2549dcb04
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4285/220073/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4285/220073/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%407f9bf99e3842.2024-05-23_07.37.39/index.html)
**OK: 456** / **Failed: 1** / **User-skipped: 41** / Auto-skipped: 0 
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_prefs_set_cdata_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_cdata_request/1-fun-0-',2,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3457}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4285/220073/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%407f9bf99e3842.2024-05-23_07.37.39/big_tests.tests.mam_SUITE.logs/run.2024-05-23_07.37.41/mam_suite.muc_prefs_set_cdata_request.7170.html)
</details>

**Review by chrzaszcz — APPROVED** [review]
Looks good 👌

**Pull request merged**